pkg/reboot/command.go

```diff
@@ -1,25 +1,32 @@
 package reboot
 
 import (
+	"github.com/google/shlex"
 	"github.com/kubereboot/kured/pkg/util"
 	log "github.com/sirupsen/logrus"
 )
 
-// CommandRebootMethod holds context-information for a command reboot.
-type CommandRebootMethod struct {
-	nodeID        string
-	rebootCommand []string
+// CommandRebooter holds context-information for a reboot with command
+type CommandRebooter struct {
+	RebootCommand []string
 }
 
-// NewCommandReboot creates a new command-rebooter which needs full privileges on the host.
-func NewCommandReboot(nodeID string, rebootCommand []string) *CommandRebootMethod {
-	return &CommandRebootMethod{nodeID: nodeID, rebootCommand: rebootCommand}
+// Reboot triggers the reboot command
+func (c CommandRebooter) Reboot() {
+	log.Infof("Invoking command: %s", c.RebootCommand)
```

Comment:
I don't know how this would ever occur but because `RebootCommand` is an `[]string` there could be a nil panic here if `RebootCommand` is nil

Reply:
I will ensure that in the constructor.